--- nagios-plugin-check_solr/check_solr.sh	2012-05-29 14:00:40.333361738 +0300
+++ nagios-plugin-check_solr-2010.10.11/check_solr.sh	2012-05-29 14:20:18.919910459 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # ============================== Summary =======================================
 # Program : check_solr.sh
 # Version : 2010.10.12
@@ -13,7 +13,7 @@
 # Notice: The plugin requires 'curl' and 'xmlstarlet' utilities in order to work
 # ================================ Change log ==================================
 # Legend:
-#                [*] Informational, [!] Bugix, [+] Added, [-] Removed
+#                [*] Informational, [!] Bugfix, [+] Added, [-] Removed
 #
 # Ver 2010.10.12:
 #                [*] Replication check was improved a bit to avoid false alarms
@@ -83,7 +83,7 @@
     echo "$SCRIPT_NAME - Nagios plugin to check Apache Solr"
     echo "version $SCRIPT_VERSION, by Alex Simenduev, http://www.planetit.ws"
 
-    echo -e "\n$USAGE\n"
+    print "\n$USAGE\n"
 
     echo "Parameters description:"
     echo " -H|--host <host>          # Solr host (default is localhost)"
@@ -140,7 +140,7 @@
 	local OUTPUT=$(exec_curl "${URL_PREFIX}${O_SOLR_HOST}:${O_SOLR_PORT}/solr/$1/replication?command=details&wt=xml" |
 		xmlstarlet sel -t -v "/response/lst[@name='details']/str[@name='isSlave']")
 
-	[ "$OUTPUT" == "true" ] && return 0 || return 1
+	[ "$OUTPUT" = "true" ] && return 0 || return 1
 }
 
 # Get ping status of specified core
@@ -154,7 +154,7 @@
 	local RESULT=$(exec_curl ${URL_PREFIX}${O_SOLR_HOST}:${O_SOLR_PORT}/solr/$1/admin/ping?wt=xml |
 		xmlstarlet sel -t -v "/response/str[@name='status']")
 
-	if [ "$RESULT" == "OK" ]; then
+	if [ "$RESULT" = "OK" ]; then
 	    echo "OK"
 	    return $STATE_OK
     else
@@ -216,7 +216,7 @@
 
         # Return CRITICAL if slave isn't currently replicating
         # and it's behind master for more then 1 hour (configurable)
-      	if [ "$SLAVE_REPLICATING" == "false" -a $SLAVE_LASTREPLICATED -gt $SOLR_MAX_SECONDS_BEHIND_MASTER ]; then
+      	if [ "$SLAVE_REPLICATING" = "false" -a $SLAVE_LASTREPLICATED -gt $SOLR_MAX_SECONDS_BEHIND_MASTER ]; then
       	    echo "${SLAVE_LASTREPLICATED} seconds behind master => CRITICAL"
       	    return $STATE_CRITICAL
         fi
@@ -243,7 +243,7 @@
 }
 
 # Print help if requested
-if [ "$1" == "--help" ] || [ "$1" == "-h" ]; then
+if [ "$1" = "--help" ] || [ "$1" = "-h" ]; then
     print_help
     exit $STATE_UNKNOWN
 elif [ $# -lt 2 ]; then
@@ -300,7 +300,7 @@
 
 # Check the metrics
 # ping
-if [ "$O_SOLR_METRIC" == "ping" ]; then
+if [ "$O_SOLR_METRIC" = "ping" ]; then
     NAGIOS_OUTPUT="Ping -"
 
     for core in $O_SOLR_CORE; do
@@ -310,7 +310,7 @@
         NAGIOS_OUTPUT="$NAGIOS_OUTPUT core '$core' $RESULT,"
     done
 # replication
-elif [ "$O_SOLR_METRIC" == "replication" ]; then
+elif [ "$O_SOLR_METRIC" = "replication" ]; then
     NAGIOS_OUTPUT="Replication -"
 
     for core in $O_SOLR_CORE; do
@@ -320,7 +320,7 @@
         NAGIOS_OUTPUT="$NAGIOS_OUTPUT core '$core' $RESULT,"
     done
 # numdocs
-elif [ "$O_SOLR_METRIC" == "numdocs" ]; then
+elif [ "$O_SOLR_METRIC" = "numdocs" ]; then
     NAGIOS_OUTPUT="Number of documents -"
     NAGIOS_PERF_OUTPUT=" |"
 
